remotes/docker/authorizer.go

```diff
@@ -284,7 +284,14 @@ func (ah *authHandler) doBearerAuth(ctx context.Context) (token string, err erro
 				// Registries without support for POST may return 404 for POST /v2/token.
 				// As of September 2017, GCR is known to return 404.
 				// As of February 2018, JFrog Artifactory is known to return 401.
-				if (errStatus.StatusCode == 405 && to.Username != "") || errStatus.StatusCode == 404 || errStatus.StatusCode == 401 {
+				// As of December 2019, Harbor is known to return 403, and it's fixed soon
+				// in https://github.com/goharbor/harbor/pull/10362
+				// As of January 2021, Docker community's registry implementation (https://github.com/distribution/distribution)
```

Comment:
Where did you see this? That implementation does not have a token server other than https://github.com/distribution/distribution/tree/main/contrib/token-server which implements the `POST` endpoint.